crates/net/network/src/discovery.rs

```diff
@@ -96,6 +103,23 @@ impl Discovery {
         })
     }
 
+    pub fn register_listener(&mut self) -> mpsc::Receiver<DiscoveryEvent> {
```

Comment:
now we need to call this from the NetworkManager and add a new networkhandlemessage variant

just like we do for this

https://github.com/paradigmxyz/reth/blob/dbafe23cce45917a43f6640d71a0216fe3268428/crates/net/network/src/network.rs#L292-L293